tests/hil/tests/ota/test_ota.py

```diff
@@ -0,0 +1,189 @@
+    "GOLIOTH_OTA_REASON_FIRMWARE_UPDATED_SUCCESSFULLY",
+    "GOLIOTH_OTA_REASON_NOT_ENOUGH_FLASH_MEMORY",
+    "GOLIOTH_OTA_REASON_OUT_OF_RAM",
+    "GOLIOTH_OTA_REASON_CONNECTION_LOST",
+    "GOLIOTH_OTA_REASON_INTEGRITY_CHECK_FAILURE",
+    "GOLIOTH_OTA_REASON_UNSUPPORTED_PACKAGE_TYPE",
+    "GOLIOTH_OTA_REASON_INVALID_URI",
+    "GOLIOTH_OTA_REASON_FIRMWARE_UPDATE_FAILED",
+    "GOLIOTH_OTA_REASON_UNSUPPORTED_PROTOCOL"
+    ]
+
+GOLIOTH_OTA_STATE_DOWNLOADING = 1
+GOLIOTH_OTA_REASON_READY = 0
+GOLIOTH_OTA_STATE_CNT = len(golioth_ota_state)
+GOLIOTH_OTA_REASON_CNT = len(golioth_ota_reason)
+TEST_BLOCK_CNT = 42
+
+pytestmark = pytest.mark.anyio
+
+@pytest.fixture(autouse=True, scope="module")
+async def setup(board, device):
+    # Delay to give NVS time to initialize
+    time.sleep(4)
+
+    # Set Golioth credentials
+    golioth_cred = (await device.credentials.list())[0]
+    board.set_golioth_psk_credentials(golioth_cred.identity, golioth_cred.key)
+
+    # Reset board
+    board.reset()
+
+    # Confirm connection to Golioth
+    assert None != board.wait_for_regex_in_line('Golioth CoAP client connected', timeout_s=60)
+
+@pytest.fixture(scope="module")
+async def tag(project, device):
+    tag_name = device.name.lower().replace('-','_')
+    tag = await project.tags.create(tag_name)
+
+    await device.add_tag(tag.id)
+
+    yield tag
+
+    try:
+        await device.remove_tag(tag.id)
+    except:
+        pass
+
+    await project.tags.delete(tag.id)
+
+@pytest.fixture(scope="module")
+async def artifacts(project):
+    artifacts = dict()
+
+    # Find Artifact that matches this device and desired update versions
+
+    all_artifacts = await project.artifacts.get_all()
+    for a in all_artifacts:
+        if a.blueprint == None:
+            if a.version == DUMMY_VER_OLDER and a.package == UPDATE_PACKAGE:
+                artifacts["test_manifest"] = a
+            elif a.version == DUMMY_VER_SAME and a.package == UPDATE_PACKAGE:
+                artifacts["test_blocks"] = a
+            elif a.version == DUMMY_VER_UPDATE and a.package == UPDATE_PACKAGE:
+                artifacts["test_reasons"] = a
+
+    assert len(artifacts) == 3
+
+    yield artifacts
+
+
+@pytest.fixture(scope="module")
+async def releases(project, artifacts, tag):
+    releases = dict()
+
+    releases["test_manifest"] = await project.releases.create([artifacts["test_manifest"].id], [], [tag.id], False)
+    releases["test_blocks"] = await project.releases.create([artifacts["test_blocks"].id], [], [tag.id], False)
+    releases["test_reasons"] = await project.releases.create([artifacts["test_reasons"].id], [], [tag.id], False)
+    yield releases
+
+    # Clean Up
+
+    for r in releases:
+        try:
+            await project.releases.delete(releases[r].id)
+        except:
+            print(f"Failed to delete release: {releases[r].id}")
+            pass
+
+@pytest.fixture(scope="function")
+async def releases_teardown(project, releases):
```

Comment:
This is running before each test, instead of after. The effect is probably the same, but the name is a little misleading.

Also, since this isn't actually returning anything and all the tests are using this fixture, we can set `autouse=True` in the decorator, and remove the `releases_teardown` argument from each test.